tools/mock-service-host/src/mid/oav/swaggerMocker.ts

```diff
@@ -55,7 +55,12 @@ export default class SwaggerMocker {
         let resourceType = ''
         let foundProvider = false
         for (let i = 0; i < pathElements.length; i++) {
-            if (i % 2 === 0 && pathElements[i].match(/microsoft\..+/i)) {
+            if (
+                i > 0 &&
```

Comment:
You may check whether we should check the resourceprovider name with desc order.  Some paths may contain multiple provider names.

Reply:
Hi @lirenhe, added logic to use the last provider.